docs/use/formatting_outputs.md

```diff
@@ -0,0 +1,218 @@
+    format_version: '0.8'
+    jupytext_version: '1.4.1'
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(use/format)=
+# Formatting code outputs
+
+(use/format/priority)=
+## Render priority
+
+When Jupyter executes a code cell it can produce multiple outputs, and each of these outputs can contain multiple [MIME media types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types), for use by different output formats (like HTML or LaTeX).
+
+MyST-NB stores a default priority dictionary for most of the common [Sphinx builders](https://www.sphinx-doc.org/en/master/usage/builders/index.html), which you can be also update in your `conf.py`.
+For example, this is the default priority list for HTML:
+
+```python
+nb_render_priority = {
+  "html": (
+            "application/vnd.jupyter.widget-view+json",
+            "application/javascript",
+            "text/html",
+            "image/svg+xml",
+            "image/png",
+            "image/jpeg",
+            "text/markdown",
+            "text/latex",
+            "text/plain",
+        )
+}
+```
+
+:::{seealso}
+[](use/format/cutomise), for a more advanced means of customisation.
+:::
+
+## Removing stdout and stderr
+
+In some cases you may not wish to display stdout/stderr outputs in your final documentation,
+for example, if they are only for debugging purposes.
+You can tell MyST-NB to remove these outputs using the `remove-stdout` and `remove-stderr` [cell tags](https://jupyter-notebook.readthedocs.io/en/stable/changelog.html#cell-tags), like so:
+
+````md
+```{code-cell} ipython3
+:tags: [remove-input,remove-stdout,remove-stderr]
+
+import pandas, sys
+print("this is some stdout")
+print("this is some stderr", file=sys.stderr)
+# but what I really want to show is:
+pandas.DataFrame({"column 1": [1, 2, 3]})
+```
+````
+
+```{code-cell} ipython3
+:tags: [remove-input,remove-stdout,remove-stderr]
+
+import pandas, sys
+print("this is some stdout")
+print("this is some stderr", file=sys.stderr)
+# but what I really want to show is:
+pandas.DataFrame({"column 1": [1, 2, 3]})
+```
+
+(use/format/images)=
+## Images
+
+With the default renderer, for any image types output by the code, we can apply formatting *via* cell metadata.
+The keys should be placed under `myst`, then for the image we can apply all the variables of the standard [image directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#image):
+
+- **width**: length or percentage (%) of the current line width
+- **height**: length
+- **scale**: integer percentage (the "%" symbol is optional)
+- **align**: "top", "middle", "bottom", "left", "center", or "right"
+- **classes**: space separated strings
+- **alt**: string
+
+Units of length are: 'em', 'ex', 'px', 'in', 'cm', 'mm', 'pt', 'pc'
+
+We can also set a caption (which is rendered as [CommonMark](https://commonmark.org/)) and name, by which to reference the figure:
+
+````md
+```{code-cell} ipython3
```

Comment:
@chrisjsewell what happens if the code in this `code-block` doesn't produce an `image`? Do we throw an error if the returned `mime` type of the `output` block isn't compatible with the `image` directive?

Reply:
Then the `render_image` method is never called, and the metadata just isn't used,
i.e. the metadata is accessed (and validated) "lazily" when it is required

Reply:
If it is invalid, for example I give a bad width here, then it produces a warning, and since this a text-based notebook, that warning will point to the correct line of the document (the first one of the code cell)

````md
```{code-cell} ipython3
---
myst:
  image:
    width: abc
---
from IPython.display import Image
Image("images/fun-fish.png")
```
````

```
/Users/chrisjsewell/Documents/GitHub/MyST-NB-actual/docs/use/formatting_outputs.md:110: WARNING: output render: Invalid image attribute: (key: 'width'; value: abc)
not a positive measure of one of the following units:
"em" "ex" "px" "in" "cm" "mm" "pt" "pc" "%"
```

and also, because the image is no longer created:

```
/Users/chrisjsewell/Documents/GitHub/MyST-NB-actual/docs/use/formatting_outputs.md:124: WARNING: 'myst' reference target not found: fun-fish
```

Reply:
I see -- thanks @chrisjsewell